lib/connect.c

```diff
@@ -1308,6 +1308,10 @@ static CURLcode singleipconnect(struct connectdata *conn,
  * pointer with the connected socket.
  */
 
+#ifndef AF_INET6
+#define AF_INET6 10
```

Comment:
Is this the best way, should we really be defining it if it's not defined. Like on mine it's 23. I worry it would lead to bugs if an include is forgotten, consider this instead

~~~diff
diff --git a/lib/connect.c b/lib/connect.c
index 5f23b79..2e25a94 100644
--- a/lib/connect.c
+++ b/lib/connect.c
@@ -1334,8 +1334,13 @@ CURLcode Curl_connecthost(struct connectdata *conn,  /* context */
 
   conn->tempfamily[0] = conn->tempaddr[0]?
     conn->tempaddr[0]->ai_family:0;
+#ifdef ENABLE_IPV6
   conn->tempfamily[1] = conn->tempfamily[0] == AF_INET6 ?
     AF_INET : AF_INET6;
+#else
+  conn->tempfamily[1] = AF_INET;
+#endif
+
   ainext(conn, 1, FALSE); /* assigns conn->tempaddr[1] accordingly */
 
   DEBUGF(infof(data, "family0 == %s, family1 == %s\n",
~~~

Reply:
I'm with @jay! This doesn't look really portable

Reply:
I took the `#define` route since
1. it doesn't add any #ifdef maze
2. since there's no IPv6 support anyway it doesn't have to be "portable", it just has to define something to a value that won't be used anyway since IPv6 doesn't work on such a build.

Reply:
Oh, and:

3. I wasn't sure how the logic will work when both families are set to IPv4. Did you try/read to verify?

Reply:
I didn't... I tried just now and it did not work the way I expected, I blocked port 80 and tried http://yahoo.com and curl ipv4-only build tried every IPv4 address twice. How about this, assuming AF_UNSPEC is always available:
~~~diff
diff --git a/lib/connect.c b/lib/connect.c
index 5f23b79..7f33ea4 100644
--- a/lib/connect.c
+++ b/lib/connect.c
@@ -1334,8 +1334,12 @@ CURLcode Curl_connecthost(struct connectdata *conn,  /* context */
 
   conn->tempfamily[0] = conn->tempaddr[0]?
     conn->tempaddr[0]->ai_family:0;
+#ifdef ENABLE_IPV6
   conn->tempfamily[1] = conn->tempfamily[0] == AF_INET6 ?
     AF_INET : AF_INET6;
+#else
+  conn->tempfamily[1] = AF_UNSPEC;
+#endif
   ainext(conn, 1, FALSE); /* assigns conn->tempaddr[1] accordingly */
 
   DEBUGF(infof(data, "family0 == %s, family1 == %s\n",
~~~

Reply:
Right, let's go with that...